steps.tf

```diff
@@ -26,14 +26,30 @@ resource "aws_s3_bucket_object" "generate_analytical_dataset_script" {
   )
 }
 
+resource "aws_s3_bucket_object" "generate_dataset_from_htme_script" {
+  bucket = data.terraform_remote_state.common.outputs.config_bucket.id
+  key    = "component/analytical-dataset-generation/generate_dataset_from_htme.py"
+  content = templatefile("${path.module}/steps/generate_dataset_from_htme.py",
+  {
+    secret_name        = local.secret_name
+    published_db       = "analytical_dataset_generation"
+    file_location      = "analytical-dataset"
+    url                = format("%s/datakey/actions/decrypt", data.terraform_remote_state.crypto.outputs.dks_endpoint[local.environment])
```

Comment:
`url` is a bit generic. Maybe `dks_endpoint`? I'm happy for that change to be made in due course though.

Reply:
will make a note of it and change later